benchmarking/factual_consistency_de.ipynb

```diff
@@ -0,0 +1,235 @@
+    "qags_xsum_scores = [item['scores']['consistency'] for item in qags_xsum_data]\n",
+    "\n",
+    "qags_cnndm_generated_outputs = [\n",
+    "    item['system_output'] for item in qags_cnndm_data\n",
+    "]\n",
+    "qags_cnndm_sources = [item['source'] for item in qags_cnndm_data]\n",
+    "qags_cnndm_scores = [item['scores']['consistency'] for item in qags_cnndm_data]\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from scipy.stats import spearmanr, pearsonr, kendalltau\n",
+    "\n",
+    "\n",
+    "def compute_correlation_values(result, annotated_scores):\n",
+    "    '''Function to compute and output the correlation values between the metric\n",
+    "    score and the human annotation scores.'''\n",
+    "    # Ignore any data points where the evaluator returned `None`. This may happen\n",
+    "    # if, for example, the prompt triggers Azure OpenAI's content filter.\n",
+    "    result_df = result.to_df()\n",
+    "    indices = list(result_df[result_df['metric_value'].notna()].index)\n",
+    "    valid_metric_values = [result.metric_values[i] for i in indices]\n",
+    "    valid_annotated_scores = [annotated_scores[i] for i in indices]\n",
+    "\n",
+    "    pearson_corr = pearsonr(valid_metric_values, valid_annotated_scores)[0]\n",
+    "    spearman_corr = spearmanr(valid_metric_values, valid_annotated_scores)[0]\n",
+    "    kendalltau_corr = kendalltau(valid_metric_values, valid_annotated_scores)[0]\n",
+    "\n",
+    "    print(f'Pearson correlation = {pearson_corr}')\n",
+    "    print(f'Spearman correlation = {spearman_corr}')\n",
+    "    print(f'Kendall-Tau correlation = {kendalltau_corr}')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Compute the factual consistency scores on QAGS-XSUM using the local (UniEval)\n",
+    "# model option and measure various correlations with the human annotated scores\n",
+    "from langcheck.metrics.de import factual_consistency\n",
+    "\n",
+    "result = factual_consistency(qags_xsum_generated_outputs, qags_xsum_sources)\n",
+    "compute_correlation_values(result, qags_xsum_scores)\n",
+    "\n",
+    "# RUN-DATE: 2024-1-17\n",
+    "# Resulting correlation values:\n",
+    "# Pearson correlation = 0.40358016311552586\n",
+    "# Spearman correlation = 0.37558373934197853\n",
+    "# Kendall-Tau correlation = 0.3097142857142857\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Compute the factual consistency scores on QAGS-XSUM using the OpenAI\n",
+    "# (gpt-3.5-turbo) model option and measure various correlations with the human\n",
+    "# annotated scores\n",
+    "from langcheck.metrics.de import factual_consistency\n",
+    "import os\n",
+    "\n",
```

Comment:
I think we need these lines below, since the model_type is `azure_openai`
```
os.environ["AZURE_OPENAI_KEY"] = 'YOUR_AZURE_OPENAI_KEY'
os.environ["OPENAI_API_VERSION"] = 'YOUR_OPENAI_API_VERSION'
os.environ["AZURE_OPENAI_ENDPOINT"] = 'YOUR_AZURE_OPENAI_ENDPOINT'
```

Reply:
oh yes, I guess I had removed it while tinkering... re-added